src/handlers/motions/motionStaked/helpers.ts

```diff
@@ -182,3 +182,71 @@ export const getUserMinStake = (
     .div(BigNumber.from(10).pow(36))
     .toString();
 };
+
+interface Props {
+  motionData: MotionData,
+  messages: MotionMessage[];
+  requiredStake: BigNumber;
+  motionStakes: MotionStakes,
+  messageKey: string,
+  vote: BigNumber,
+  staker: string,
+  amount: BigNumber,
+}
+
+export const getUpdatedMessages = ({
+  motionData,
+  messages,
+  requiredStake,
+  motionStakes,
+  messageKey,
+  vote,
+  staker,
+  amount,
+}: Props): MotionMessage[] => {
+  const updatedMessages = [...messages];
+  const isFirstObjection = vote.eq(MotionVote.NAY) && !motionData.hasObjection;
+  if (isFirstObjection) {
+    motionData.hasObjection = true;
+    updatedMessages.push(
+      {
+        name: 'ObjectionRaised',
+        messageKey: `${messageKey}_ObjectionRaised`,
+        initiatorAddress: staker,
+      },
+    );
+  }
+
+  updatedMessages.push(
+    {
+      name: 'MotionStaked',
+      messageKey,
+      initiatorAddress: staker,
+      vote: vote.toString(),
+      amount: amount.toString(),
+    },
+  );
+
+  if (vote.eq(MotionVote.YAY) && requiredStake.eq(motionStakes.raw.yay)) {
+    const messageName = motionData.hasObjection ? 'MotionFullyStakedAfterObjection' : 'MotionFullyStaked';
+    updatedMessages.push(
+      {
+        name: messageName,
+        messageKey: `${messageKey}_${messageName}`,
+        initiatorAddress: staker,
+      },
+    );
+  }
+
+  if (vote.eq(MotionVote.NAY) && requiredStake.eq(motionStakes.raw.nay) && BigNumber.from(motionStakes.raw.nay).gt(motionStakes.raw.yay)) {
+    updatedMessages.push(
```

Comment:
How come we have this check here also?:

`BigNumber.from(motionStakes.raw.nay).gt(motionStakes.raw.yay))`

Aren't these are sufficient to determine if it's fully staked?

``` js
vote.eq(MotionVote.NAY) && requiredStake.eq(motionStakes.raw.nay)
```

Reply:
In any case, a comment in the code would be useful here

Reply:
@willm30 I added this based on your comment in the cdapp PR that we shouldn't show ObjectionFullyStaked if the motion is already fully staked in favor. `BigNumber.from(motionStakes.raw.nay).gt(motionStakes.raw.yay)` is the same logic in the dapp to handle this situation. The only way the nay vote could be greater than the yay vote is if the motion was not already fully staked for the yay vote. I can add a comment to explain.